### PR TITLE
Coverting sample-pipeline to cue

### PR DESCRIPTION
**Pull request by pxp928**
Took some time to debug but the issue was that the kaniko task did not have the required `IMAGE_DIGEST` and `IMAGE_URL` that allowed it to get signed. The new cue implementation fixes this and works with Kyverno verification. Works with the older current version of chains. I also tested with the newest version to determine the error output as the current version of chains did not throw any errors.....

### CODE REVIEW

File: examples/buildpacks/buildpacks.cue [hunk 1]
```diff
@@ -34,3 +34,18 @@ pipelineRun: "cache-image-pipelinerun-": spec: {
...
+			accessModes: ["ReadWriteOnce"]
+			resources: requests: storage: "500Mi"
+		}
+	}
+}
+
+pipelineRun: [Name=_]: spec: workspaces: [{
+	name: *"\(Name)ws" | string
+	persistentVolumeClaim: claimName: "\(Name)source-ws-pvc"
+}, ...]
```
Comment (bradbeck) [inline examples/buildpacks/buildpacks.cue:51]: I think moving this here breaks the `ibm-tutorial` pipeline run.

Comment (pxp928) [reply examples/buildpacks/buildpacks.cue:51]: You are right. Is there another place to move it? It causes issues with the sample-pipeline.

Comment (pxp928) [reply examples/buildpacks/buildpacks.cue:51]: I think i got it. I will revise

Comment (bradbeck) [reply examples/buildpacks/buildpacks.cue:51]: See https://github.com/pxp928/ssf/pull/1

Comment (pxp928) [reply examples/buildpacks/buildpacks.cue]: fixed. Moved back to example.cue

Comment (pxp928) [reply examples/buildpacks/buildpacks.cue]: Yup thats exactly what i did

File: examples/sample-pipeline/sample-pipeline.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+set -euo pipefail
+
+# Define variables.
+GIT_ROOT=$(git rev-parse --show-toplevel)
+DEFAULT_REPOSITORY=$(xxd -l 16 -c 16 -p < /dev/random)
+: "${REGISTRY:=ttl.sh}"
+: "${REPOSITORY:=$REGISTRY/$DEFAULT_REPOSITORY}"
+C_GREEN='\033[32m'
+C_YELLOW='\033[33m'
+C_RED='\033[31m'
```
Comment (bradbeck) [inline examples/sample-pipeline/sample-pipeline.sh:11]: These could probably be removed to make the shell linter happy.

Comment (pxp928) [reply examples/sample-pipeline/sample-pipeline.sh]: done!

**Comment by bradbeck** [comment]
Maybe we should add `example-sample-pipeline` to the `Makefile`.

**Comment by pxp928** [comment]
> Maybe we should add `example-sample-pipeline` to the `Makefile`.

Updated README and added command to Makefile.

File: examples/sample-pipeline/README.md [hunk 3]
```diff
@@ -43,41 +27,36 @@ You can verify these definitions using [tapstry-pipelines](https://github.com/ta
...
 
-In this example, we are using [tekton-tutorial-openshift](https://github.com/IBM/tekton-tutorial-openshift) application sample application. If you want to use other application, make sure to make corresponding changes to `ssf-run.yaml`. You may also have to change params to `build` and `deploy` tasks to reflect correct location for `Dockerfile` context and deployment file resp.
+# Run a new pipeline.
+make example-sample-pipeline
 
-```bash
-% kubectl create -f ssf-run.yaml
-```
+# Export the value of APP_IMAGE from the pipelinerun describe as DOCKER_IMG:
+export DOCKER_IMG=$(tkn pr describe --last -o jsonpath='{.spec.params[?(@.name=="APP_IMAGE")].value}')
```
Comment (bradbeck) [inline examples/sample-pipeline/README.md:43]: I don't think this is quite right. I think it's more like what is in `ibm-tutorials`. e.g. 
```bash
$ tkn pr describe --last -o jsonpath='{.spec.params}' | jq
[
  {
    "name": "gitUrl",
    "value": "https://github.com/IBM/tekton-tutorial-openshift"
  },
  {
    "name": "pathToYamlFile",
    "value": "kubernetes/picalc.yaml"
  },
  {
    "name": "imageUrl",
    "value": "192.168.1.37:8888/hello-ssf"
  },
  {
    "name": "imageTag",
    "value": "1h"
  }
]
```
Doesn't contain `APP_IMAGE`.

File: examples/sample-pipeline/README.md [hunk 4]
```diff
@@ -43,41 +27,36 @@ You can verify these definitions using [tapstry-pipelines](https://github.com/ta
...
+tkn pr logs --last -f
 
-To observe execution of the pipeline, consider setting up [tektoncd/dashboard](https://github.com/tektoncd/dashboard). 
-The SBOM and vulnerability-report are accessible only in the task logs. 
+# Double check that the attestation and the signature were uploaded to the OCI.
+crane ls "${DOCKER_IMG}"
+
+# Verify the image and the attestation.
+cosign verify --key k8s://tekton-chains/signing-secrets "${DOCKER_IMG}"
+cosign verify-attestation --key k8s://tekton-chains/signing-secrets "${DOCKER_IMG}"
```
Comment (bradbeck) [inline examples/sample-pipeline/README.md:53]: Something like this should work:
```
export IMAGE_URL=$(tkn pr describe --last -o jsonpath='{.spec.params[?(@.name=="imageUrl")].value}')
export IMAGE_TAG=$(tkn pr describe --last -o jsonpath='{.spec.params[?(@.name=="imageTag")].value}')
export DOCKER_IMG="${IMAGE_URL}:${IMAGE_TAG}"

# Wait until it completes.
tkn pr logs --last -f

# Ensure it has been signed.
tkn tr describe --last -o jsonpath='{.metadata.annotations.chains\.tekton\.dev/signed}'
# Should output "true"

# Double check that the attestation and the signature were uploaded to the OCI.
crane ls "${IMAGE_URL}"

# Verify the image and the attestation.
cosign verify --key k8s://tekton-chains/signing-secrets "${DOCKER_IMG}"
cosign verify-attestation --key k8s://tekton-chains/signing-secrets "${DOCKER_IMG}"
```

File: examples/sample-pipeline/README.md (hunk 3 above)
Comment (pxp928) [reply examples/sample-pipeline/README.md]: oops. Fixed to use `imageUrl`

File: Makefile [hunk 5]
```diff
@@ -57,6 +57,10 @@ setup-opa-gatekeeper: ##  Setup opa gatekeeper
 example-buildpacks: ## Run the buildpacks example
 	bash examples/buildpacks/buildpacks.sh
 
+.PHONY: example-sample-pipeline
+example-sample-pipeline: ## Run the buildpacks example
```
Comment (bradbeck) [inline Makefile:61]:
```suggestion
example-sample-pipeline: ## Run the sample-pipeline example
```

**Review by bradbeck — APPROVED** [review]

**Pull request merged**